tests/core/pyspec/eth2spec/test/helpers/attestations.py

```diff
@@ -373,6 +374,26 @@ def temp_participants_filter(comm):
     return attestations
 
 
+_prep_state_cache_dict = LRU(size=10)
+
+
+def cached_prepare_state_with_attestations(spec, state):
+    """
+    Cached version of prepare_state_with_attestations,
+    but does not return anything, and does not support a participation fn argument
+    """
+    # If the pre-state is not already known in the LRU, then take it, make it leaking, and put it in the LRU.
+    # The input state is likely already cached, so the hash-tree-root is fine.
```

Comment:
fine wrt speed, right? Should clarify. If I came to this comment 4 months from now, I'd be nervous

Reply:
Yes, wrt speed. A hash-tree-root of the state is not scare when the state is used in a lot of transitions, and the work is nicely cached in the tree internally.